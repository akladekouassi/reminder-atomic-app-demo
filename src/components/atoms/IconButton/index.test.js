import { shallow, render } from "enzyme";

import DeleteIcon from "@material-ui/icons/Delete";
import IconButton from ".";

describe("Icon Button", () => {
  it("Renders Primary Button", () => {
    const wrapper = render(<IconButton icon={<DeleteIcon />} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it("Renders Disabled Button", () => {
    const wrapper = render(<IconButton icon={<DeleteIcon />} disabled />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it("Handles button click", () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(
      <IconButton icon={<DeleteIcon />} onClick={mockFunction} />
    );
    wrapper.simulate("click");
    wrapper.update();
    expect(mockFunction).toBeCalled();
  });
});
