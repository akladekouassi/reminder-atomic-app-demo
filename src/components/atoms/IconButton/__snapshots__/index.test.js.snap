// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Button Renders Disabled Button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary Mui-disabled Mui-disabled",
      "disabled": "",
      "tabindex": "-1",
      "type": "button",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiIconButton-label",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "MuiSvgIcon-root",
              "focusable": "false",
              "viewBox": "0 0 24 24",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "d": "M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "focusable": undefined,
              "viewBox": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "focusable": undefined,
              "viewBox": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "disabled": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "disabled": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Primary Button Renders Primary Button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary",
      "tabindex": "0",
      "type": "button",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiIconButton-label",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "MuiSvgIcon-root",
              "focusable": "false",
              "viewBox": "0 0 24 24",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "d": "M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "focusable": undefined,
              "viewBox": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "focusable": undefined,
              "viewBox": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
